Mechaduino/Mechaduino/Controller.cpp

```diff
@@ -67,16 +67,25 @@ void TC5_Handler() {                // gets called with FPID frequency
 
     y_1 = y;  //copy current value of y to previous value (y_1) for next control cycle before PA angle added
 
+    // Depending on direction we want to apply torque, add or subtract a phase angle of PA for max effective torque.
+    // PA should be equal to one full step angle: if the excitation angle is the same as the current position, we would not move!
+    // If we are close to target (u is close to 0), phase angle is adjusted to intermediate value to reduce oscillations.
+    float pa = PA;
+
+    if (u > -deadband && u < deadband)
+    {
+      pa = PA * abs(u) / deadband;
```

Comment:
It might be a good idea calculate inverse of deadband once where its defined, and multiply by this inversion. I think doing an expensive division every interrupt is not good practice (Armv6 doesn't have division instruction).

Reply:
Possibly true, but it is a float division so a float multiplication is only slightly faster.

Reply:
Sorry you are correct, I forgot about the lack of floating point hardware. Out of curiosity I will try both approaches and time the results.

Reply:
> 
> 
> Sorry you are correct, I forgot about the lack of floating point hardware. Out of curiosity I will try both approaches and time the results.

I just had some time to test, your approach took 20uS, the inverted multiplication 12uS. The code seemed to work fine.